Casks/y/yacreader.rb

```diff
@@ -9,8 +9,16 @@
   homepage "https://www.yacreader.com/"
 
   livecheck do
-    url "https://www.yacreader.com/downloads"
-    regex(%r{href=.*?/YACReader[._-]v?(\d+(?:\.\d+)+)[._-]MacOSX[._-]U[._-]Qt6\.dmg}i)
+    url :url
+    regex(/YACReader[._-]v?(\d+(?:\.\d+)+)[._-]MacOSX[._-]U[._-]Qt6\.dmg/i)
+    strategy :github_latest do |json, regex|
+      json["assets"]&.map do |asset|
+        match = asset["browser_download_url"]&.match(regex)
```

Comment:
```suggestion
    regex(/^YACReader[._-]v?(\d+(?:\.\d+)+)[._-]MacOSX[._-]U[._-]Qt6\.dmg$/i)
    strategy :github_latest do |json, regex|
      json["assets"]&.map do |asset|
        match = asset["name"]&.match(regex)
```

Since we are only matching parts of the filename, we can check the asset name directly.